### PR TITLE
Convert array to linked list to support dynamic endpoints

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* Static array in onff cluster and identify cluster did not support dynamic endpoints

#### Change overview
* Changed array to linked list

#### Testing
* Automated tests to check for regressions and to verify changes
* Manual testing to validate changes

### CODE REVIEW

**Review by mkardous-silabs — COMMENTED** [review]
ready to merge

**Review by woody-apple — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/clusters/identify-server/identify-server.h [hunk 1]
```diff
@@ -58,5 +58,21 @@ struct Identify
     EmberAfIdentifyEffectIdentifier mCurrentEffectIdentifier;
     EmberAfIdentifyEffectIdentifier mTargetEffectIdentifier;
     uint8_t mEffectVariant;
-    bool mActive = false;
+    bool mActive             = false;
+    Identify * next_identify = nullptr;
+
+    bool hasNext()
+    {
+        bool hasNext = false;
```
Comment (Damian-Nordic) [inline src/app/clusters/identify-server/identify-server.h:66]: return `this->next_identify != nullptr`?

File: src/app/clusters/identify-server/identify-server.cpp [hunk 2]
```diff
@@ -64,42 +64,47 @@ using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters::Identify;
 
-static std::array<Identify *, EMBER_AF_IDENTIFY_CLUSTER_SERVER_ENDPOINT_COUNT> instances = { 0 };
+static Identify * first_identify = nullptr;
```
Comment (Damian-Nordic) [inline src/app/clusters/identify-server/identify-server.cpp:67]: nit: CHIP codebase uses rather camelCase naming convention.

**Review by Damian-Nordic — APPROVED** [review]
I have some optional code style suggestions, but looking good overall.

**Review by andy31415 — APPROVED** [review]

File: src/app/clusters/identify-server/identify-server.h (hunk 1 above)
Comment (mkardous-silabs) [reply src/app/clusters/identify-server/identify-server.h]: done

File: src/app/clusters/identify-server/identify-server.cpp (hunk 2 above)
Comment (mkardous-silabs) [reply src/app/clusters/identify-server/identify-server.cpp]: done

**Pull request merged**